addons/mail/models/mail_address_mixin.py

```diff
@@ -14,24 +14,22 @@ class MailAddressMixin(models.AbstractModel):
         """
     _name = 'mail.address.mixin'
     _description = 'Email address mixin'
-    _primary_email = ['email']
+    _primary_email = 'email'
 
     email_normalized = fields.Char(string='Normalized email address', compute="_compute_email_normalized", invisible=True,
                                   compute_sudo=True, store=True, help="""This field is used to search on email address,
                                   as the primary email field can contain more than strictly an email address.""")
 
-    @api.depends(lambda self: self._primary_email)
+    @api.depends(lambda self: [self._primary_email])
     def _compute_email_normalized(self):
         self._assert_primary_email()
-        [email_field] = self._primary_email
         for record in self:
-            record.email_normalized = tools.email_normalize(record[email_field])
+            record.email_normalized = tools.email_normalize(record[self._primary_email])
 
     def _assert_primary_email(self):
         if not hasattr(self, "_primary_email") or \
-                not isinstance(self._primary_email, (list, tuple)) or \
-                not len(self._primary_email) == 1:
+                not isinstance(self._primary_email, str) or \
+                not len(self._primary_email) > 1:
```

Comment:
Having more than 1 character in your string does not seem a real condition. Moreover I think I could define a field named a which could be used as a primary email ;) .